electricitylci/generation_mix.py

```diff
@@ -342,19 +339,150 @@ def olcaschema_genmix(database, gen_dict, subregion=None):
-                        f"Trouble matching dictionary for {fuelname} - {reg}"
+                        f"Trouble matching dictionary for generation mix {fuelname} - {reg}. Skipping this flow for now"
                     )
                 else:
+                    ra = exchange_table_creation_input_genmix(
+                    database_f1, fuelname
+                    )
+                    ra["quantitativeReference"] = False                    
                     ra["provider"] = {
-                        "name": matching_dict["name"],
-                        "@id": matching_dict["uuid"],
-                        "category": matching_dict["category"].split("/"),
+                       "name": matching_dict["name"],
+                       "@id": matching_dict["uuid"],
+                       "category": matching_dict["category"].split("/"),
                     }
-                exchange(ra, exchanges_list)
-                # Writing final file
+                    #if matching_dict is None:
+                    exchange(ra, exchanges_list)
+                    # Writing final file
 
         final = process_table_creation_genmix(reg, exchanges_list)
+        # print(reg +' Process Created')
+        generation_mix_dict[reg] = final
+    return generation_mix_dict
+
+
+def olcaschema_usaverage(database, gen_dict, subregion=None):
+    if subregion is None:
+        subregion = model_specs.regional_aggregation
+    generation_mix_dict = {}
+
+    us_database = create_generation_mix_process_df_from_egrid_ref_data(subregion='US')
+    #Not choosing the Hawaiian and Alaskan regions. 
+    us_database = us_database[(us_database['Subregion'] != 'HIMS') &  (us_database['Subregion'] != 'HIOA') & (us_database['Subregion'] != 'AKGD') & (us_database['Subregion'] != 'AKMS')]
+    df2 = us_database.groupby(['FuelCategory'])['Electricity'].agg('sum').reset_index()
+    df2['Electricity_fuel_total'] = df2['Electricity']
+    del df2['Electricity']
+    df3 = us_database.merge(df2,left_on ='FuelCategory',right_on = 'FuelCategory')
+    df3['Generation_Ratio'] = df3['Electricity']/df3['Electricity_fuel_total']
+    del df3['Electricity_fuel_total']
+    us_database = df3
+    if "FuelCategory" in us_database.columns:
+        fuels = list(pd.unique(us_database["FuelCategory"]))
+
+    for fuel in fuels:
+
+        database_reg = us_database[us_database["FuelCategory"] == fuel]
+        exchanges_list = []
 
+        # Creating the reference output
+        exchange(exchange_table_creation_ref(database_reg), exchanges_list)
+        for reg in list(us_database["Subregion"].unique()):
+            # Reading complete fuel name and heat content information
+            # fuelname = row['Fuelname']
+            # croppping the database according to the current fuel being considered
+            #Not choosing the Hawaiian and Alaskan regions. 
+            if reg == 'HIMS' or reg == 'HIOA' or reg == 'AKGD' or reg == 'AKMS':
```

Comment:
If we stick with the above suggestion, this could become:
`if reg in excluded_regions:`

Reply:
Will follow the suggested method.